File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const counts = tangerine.reduce((acc, cur) => {
        acc.set(cur, (acc.get(cur) || 0) + 1);
        return acc;
    }, new Map());
    
    const countsArr = [...counts.values()].sort((a, b) => b-a);
    let result = 0;
    
    for (let count of countsArr) {
        if (k >= count) {
            k -= count;
            result++;
        } else if (k > 0) {
            result++;
            break;
        } else {
            break;
        }
    }
    
    return result;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import defaultdict

def solution(k, tangerine):
    result = 0
    dict_tang = defaultdict(int)
    for i in tangerine:
        dict_tang[i] += 1
    
    dict_tang = dict(sorted(dict_tang.items(), key=lambda x: x[1], reverse=True))
    
    for key, value in dict_tang.items():
        if k <= 0:
            return result
        k -= value
        result += 1
    
    return result